site/source/docs/porting/pthreads.rst

```diff
@@ -4,29 +4,41 @@
+- Optionally, pass the linker flag ``-s PTHREAD_HINT_NUM_CORES=<integer>`` to choose what the function emscripten_num_logical_cores(); will return if navigator.hardwareConcurrency is not supported. If -1 is specified here, a popup dialog will be shown at startup to let the user specify the value that is returned here. This can be helpful in order to dynamically test how an application behaves with different values here.
+
+There should be no other changes required. In C/C++ code, the preprocessor check ``#ifdef __EMSCRIPTEN_PTHREADS__`` can be used to detect whether Emscripten is currently targeting pthreads.
+
+Additional flags
+================
 
-- Pass the compiler flag -s USE_PTHREADS=1 when compiling any .c/.cpp files, AND when linking to generate the final output .js file.
-- Optionally, pass the linker flag -s PTHREAD_POOL_SIZE=<integer> to specify a predefined pool of web workers to populate at page preRun time before application main() is called. If -1 is passed to both PTHREAD_POOL_SIZE and PTHREAD_HINT_NUM_CORES, then a popup dialog will ask the user the size of the pool (useful for testing).
-- Optionally, pass the linker flag -s PTHREAD_HINT_NUM_CORES=<integer> to choose what the function emscripten_num_logical_cores(); will return if navigator.hardwareConcurrency is not supported. If -1 is specified here, a popup dialog will be shown at startup to let the user specify the value that is returned here. This can be helpful in order to dynamically test how an application behaves with different values here.
+- ``-s PROXY_TO_PTHREAD``: In this mode your original ``main()`` is replaced by
+  a new one that creates a pthread and runs the original ``main()`` on it. As a
+  result, your application's main code is run off the main thread, which is good
+  for responsiveness. The main thread does still run code when things are
+  proxied to it, for example to handle events, rendering, etc.
 
-There should be no other changes required. In C/C++ code, the preprocessor check #ifdef __EMSCRIPTEN_PTHREADS__ can be used to detect whether Emscripten is currently targeting pthreads.
+Note that Emscripten has ``--proxy-to-worker`` which sounds similar but is
+unrelated. That flag does not use pthreads or SharedArrayBuffer, and instead
+uses a plain Web Worker to run your main program (and postMessage to proxy
+messages back and forth).
 
 Special considerations
 ======================
 
 The Emscripten implementation for the pthreads API should follow the POSIX standard closely, but some behavioral differences do exist:
 
-- At runtime, you can `#include <emscripten/threading.h>` and use the ``emscripten_has_threading_support()`` function to
+- At runtime, you can ``#include <emscripten/threading.h>`` and use the ``emscripten_has_threading_support()`` function to
```

Comment:
I think it's the same as the ifdef, yeah - the existence of the function is because there was a period where a single build could do both.

I'll remove this part and mention that no single build can do both pthreads and non pthreads.